examples/pytorch/rgcn/entity_sample.py

```diff
@@ -1,167 +1,125 @@
-
-    train_loader, val_loader, test_loader = init_dataloaders(
-        args, g, train_idx, test_idx, target_idx, args.gpu)
-
-    model = RGCN(g.num_nodes(),
-                 args.n_hidden,
-                 num_classes,
-                 num_rels,
-                 num_bases=args.n_bases,
-                 dropout=args.dropout,
-                 self_loop=args.use_self_loop,
-                 ns_mode=True)
-    labels = labels.to(device)
-    model = model.to(device)
-    inv_target = inv_target.to(device)
-
-    optimizer = th.optim.Adam(model.parameters(), lr=1e-2, weight_decay=args.wd)
-
-    for epoch in range(args.n_epochs):
-        train_acc, loss = train(model, train_loader, inv_target, labels, optimizer)
-        print("Epoch {:05d}/{:05d} | Train Accuracy: {:.4f} | Train Loss: {:.4f}".format(
-            epoch, args.n_epochs, train_acc, loss))
-
-        val_logits, val_seeds = evaluate(model, val_loader, inv_target)
-        val_acc = accuracy(val_logits.argmax(dim=1), labels[val_seeds].cpu()).item()
-        print("Validation Accuracy: {:.4f}".format(val_acc))
-
-    test_logits, test_seeds = evaluate(model, test_loader, inv_target)
-    test_acc = accuracy(test_logits.argmax(dim=1), labels[test_seeds].cpu()).item()
-    print("Final Test Accuracy: {:.4f}".format(test_acc))
-
+            eval_seeds.append(output_nodes.cpu().detach())
+    eval_logits = torch.cat(eval_logits)
+    eval_seeds = torch.cat(eval_seeds)
+    return  accuracy(eval_logits.argmax(dim=1), labels[eval_seeds].cpu()).item()
+
+def train(device, g, target_idx, labels, train_mask, model):
+    # define train idx, loss function and optimizer
+    train_idx = torch.nonzero(train_mask, as_tuple=False).squeeze()
+    loss_fcn = nn.CrossEntropyLoss()
+    optimizer = torch.optim.Adam(model.parameters(), lr=1e-2, weight_decay=5e-4)
+    # construct sampler and dataloader
+    sampler = MultiLayerNeighborSampler([4, 4])
+    train_loader = DataLoader(g, target_idx[train_idx], sampler, device=device, 
+                              batch_size=100, shuffle=True)
+    # no separate validation subset, use train index instead for validation
+    val_loader = DataLoader(g, target_idx[train_idx], sampler, device=device, 
+                            batch_size=100, shuffle=False)
+    for epoch in range(50):
+        model.train()
+        total_loss = 0
+        for it, (input_nodes, output_nodes, blocks) in enumerate(train_loader):
+            output_nodes = inv_target[output_nodes]
+            for block in blocks:
+                block.edata['norm'] = dgl.norm_by_dst(block).unsqueeze(1)
+            logits = model(blocks)
+            loss = loss_fcn(logits, labels[output_nodes])
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            total_loss += loss.item()
+        acc = evaluate(model, labels, val_loader, inv_target)
+        print("Epoch {:05d} | Loss {:.4f} | Accuracy {:.4f} "
```

Comment:
"Accuracy" -> "Val Accuracy" as you reported "Train Accuracy" in `entity.py`.

Reply:
Right. Thanks. I updated.